### PR TITLE
[front] - enh(DataSourceViewSelector): disable root node check

### PR DESCRIPTION
**Pull request by JulesBelveze**
## Description

This PR refactors the `DataSourceViewSelector` component to improve the user experience when selecting data sources. 

The main changes include:
- Removing the checkbox from the root node of the `DataSourceViewSelector` tree when updating the connection (while keeping it in the assistant builder).
- Introducing a "Select All" button as an action for the root node, replacing the previous checkbox functionality.

Screenshot 2024-09-25 at 17 29 17

**References:**
- https://github.com/dust-tt/tasks/issues/1390

## Deploy Plan

- Apply migration from https://github.com/dust-tt/dust/pull/7716 first
- Deploy `front`

Note: there will be a short time span during which people will be able to unselect/select root nodes while the backend won't accept it

### CODE REVIEW

**Comment by JulesBelveze** [comment]
Run this migration first https://github.com/dust-tt/dust/pull/7716

**Comment by spolu** [comment]
Exciting!

What's the release plan?

Do we stop creating by default? Do we create with all children by default? Then we backflll + merge?

**Comment by JulesBelveze** [comment]
I would probably create data source views with no children (empty array) as `parentsIn` rather than having all children by default. Since we will have a "Select All" button, it won't be painful for the admin to add all children if needed.

Then we run the #7716 backfill and merge this one 🤓 Wdyt?

**Comment by spolu** [comment]
Makes sense but it likely requires a "screen" at the end of connection setup to explain that we're syncing and that data can be added to the company data etc... I presume the work on that screen is planned but not sure where we're at with it?

**Comment by JulesBelveze** [comment]
Yep, I think that's the ticket for this screen: https://github.com/dust-tt/dust/issues/7361

**Comment by spolu** [comment]
Awesome let's make sure to sequence #7361 before :+1:

File: front/components/assistant_builder/AssistantBuilderDataSourceModal.tsx [hunk 1]
```diff
@@ -113,6 +113,7 @@ export default function AssistantBuilderDataSourceModal({
             selectionConfigurations={selectionConfigurations}
             setSelectionConfigurations={setSelectionConfigurationsCallback}
             viewType={viewType}
+            displayMode="assistant_builder"
```
Comment (tdraier) [inline front/components/assistant_builder/AssistantBuilderDataSourceModal.tsx:116]: nit: showSelectAll maybe more clear if it's the goal of this prop ?

**Review by tdraier — APPROVED** [review]

**Review by spolu — APPROVED** [review]
LGTM

**Review by tdraier — APPROVED** [review]
LGTM

File: front/components/data_source_view/DataSourceViewSelector.tsx [hunk 2]
```diff
@@ -388,40 +317,73 @@ export function DataSourceViewSelector({
...
+          canBeExpanded(viewType, dataSourceView.dataSource) ? "node" : "leaf"
+        }
+        checkbox={
+          !isRootSelectable
+            ? {
+                checked: checkedStatus,
+                onChange: () => {
+                  selectionConfiguration.isSelectAll =
+                    checkedStatus === "checked";
+                },
```
Comment (tdraier) [inline front/components/data_source_view/DataSourceViewSelector.tsx:357]: You can't just change a prop, you need to call the updater : 
```suggestion
                onChange: () => {
                  setSelectionConfigurations((prevState) => {
                    const prevSelectionConfiguration =
                      prevState[dataSourceView.sId] ??
                      defaultSelectionConfiguration(dataSourceView);
                    const udpatedConfig = {
                      ...prevSelectionConfiguration,
                      selectedResources: [],
                      isSelectAll: checkedStatus !== "checked",
                    };

                    return {
                      ...prevState,
                      [dataSourceView.sId]: udpatedConfig,
                    };
                  });
                },

```

Comment (JulesBelveze) [reply front/components/data_source_view/DataSourceViewSelector.tsx]: Python's habits 😅

**Pull request merged**